antarest/study/service.py

```diff
@@ -2640,3 +2641,37 @@ def asserts_no_thermal_in_binding_constraints(
             if ref_bcs:
                 binding_ids = [bc.id for bc in ref_bcs]
                 raise ReferencedObjectDeletionNotAllowed(cluster_id, binding_ids, object_type="Cluster")
+
+    def delete_file_or_folder(self, study_id: str, path: str, current_user: JWTUser) -> None:
+        """
+        Deletes a file or a folder of the study.
+        The data must be located inside the 'User' folder.
+        Also, it can not be inside the 'expansion' folder.
+
+        Args:
+            study_id: UUID of the concerned study
+            path: Path corresponding to the resource to be deleted
+            current_user: User that called the endpoint
+
+        Raises:
+            FileDeletionNotAllowed: if the path does not comply with the above rules
+        """
+        study = self.get_study(study_id)
+        assert_permission(current_user, study, StudyPermissionType.WRITE)
+
+        url = [item for item in path.split("/") if item]
+        if len(url) < 2 or url[0] != "user":
+            raise FileDeletionNotAllowed(f"the targeted data isn't inside the 'User' folder: {path}")
+        if url[1] == "expansion":
+            raise FileDeletionNotAllowed(f"you cannot delete a file/folder inside 'expansion' folder : {path}")
+
+        study_tree = self.storage_service.raw_study_service.get_raw(study, True).tree.build()
+        try:
+            study_tree["user"].delete(url[1:])
+        except ChildNotFoundError as e:
+            raise FileDeletionNotAllowed(f"the given path doesn't exist: {e.detail}")
```

Comment:
We should use "raise from e" (so that e is correctly identified as the direct cause)